anomalib/data/base.py

```diff
@@ -0,0 +1,207 @@
+        |---|-------------------|-----------|-------------|------------------|-------|
+        | 0 | path/to/image.png | anomalous | 0           | path/to/mask.png | train |
+        |---|-------------------|-----------|-------------|------------------|-------|
+        """
+        raise NotImplementedError
+
+    def get_samples(self, split: Optional[str] = None) -> DataFrame:
+        """Retrieve the samples of the full dataset or one of the splits (train, val, test).
+
+        Args:
+            split: (str): The split for which we want to retrieve the samples ("train", "val" or "test"). When
+                left empty, all samples will be returned.
+
+        Returns:
+            DataFrame: A dataframe containing the samples of the split or full dataset.
+        """
+        assert self._samples is not None, "Samples have not been created yet."
+        if split is None:
+            return self._samples
+        samples = self._samples[self._samples.split == split]
+        return samples.reset_index(drop=True)
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        """Setup train, validation and test data.
+
+        Args:
+          stage: Optional[str]:  Train/Val/Test stages. (Default value = None)
+        """
+        self._samples = self._create_samples()
+
+        logger.info("Setting up train, validation, test and prediction datasets.")
+        if stage in (None, "fit"):
+            samples = self.get_samples("train")
+            self.train_data = AnomalibDataset(
+                samples=samples,
+                split="train",
+                task=self.task,
+                pre_process=self.pre_process_train,
+            )
+
+        if stage in (None, "fit", "validate"):
+            samples = self.get_samples("val") if self.create_validation_set else self.get_samples("test")
+            self.val_data = AnomalibDataset(
+                samples=samples,
+                split="val",
+                task=self.task,
+                pre_process=self.pre_process_val,
+            )
+
+        if stage in (None, "test"):
+            samples = self.get_samples("test")
+            self.test_data = AnomalibDataset(
+                samples=samples,
+                split="test",
+                task=self.task,
+                pre_process=self.pre_process_val,
+            )
+
+    def contains_anomalous_images(self, split: Optional[str] = None) -> bool:
+        """Check if the dataset or the specified subset contains any anomalous images.
+
+        Args:
+            split (str): the subset of interest ("train", "val" or "test"). When left empty, the full dataset will be
+                checked.
+
+        Returns:
+            bool: Boolean indicating if any anomalous images have been assigned to the dataset or subset.
+        """
+        samples = self.get_samples(split)
+        return 1 in list(samples.label_index)
```

Comment:
```suggestion
        return LABEL_INDEX_ANOMALOUS in list(samples.label_index)
```
a bit picky but a constant would clearer

Reply:
In this case would using enums be better? We can then have something like `Label.Normal` and `Label.Anomalous`